contrib/to-a11y-string/to-a11y-string.js

```diff
@@ -0,0 +1,735 @@
+    "\\underleftrightarrow": "left-right arrow",
+    "\\undergroup": "group",
+    "\\underlinesegment": "line segment",
+    "\\utilde": "tilde",
+};
+
+type NestedArray<T> = Array<T | NestedArray<T>>;
+
+const buildString = (
+    str: string,
+    type: Atom | "normal",
+    a11yStrings: NestedArray<string>,
+) => {
+    if (!str) {
+        return;
+    }
+
+    let ret;
+
+    if (type === "open") {
+        ret = str in openMap ? openMap[str] : stringMap[str] || str;
+    } else if (type === "close") {
+        ret = str in closeMap ? closeMap[str] : stringMap[str] || str;
+    } else if (type === "bin") {
+        ret = binMap[str] || str;
+    } else if (type === "rel") {
+        ret = relMap[str] || str;
+    } else {
+        ret = stringMap[str] || str;
+    }
+
+
+    // If nothing was found and it's not a plain string or number
+    // if (ret === str && !/^\w+$/.test(str)) {
+    //     // This is likely a case that we'll need to handle
+    //     throw new Error("KaTeX a11y " + type + " string not found: " + str);
+    // }
+
+    // If the text to add is a number and there is already a string
+    // in the list and the last string is a number then we should
+    // combine them into a single number
+    if (
+        /^\d+$/.test(ret) &&
+        a11yStrings.length > 0 &&
+        // TODO(kevinb): check that the last item in a11yStrings is a string
+        // I think we might be able to drop the nested arrays, which would make
+        // this easier to type - $FlowFixMe
+        /^\d+$/.test(a11yStrings[a11yStrings.length - 1])
+    ) {
+        a11yStrings[a11yStrings.length - 1] += ret;
+    } else if (ret) {
+        a11yStrings.push(ret);
+    }
+};
+
+const buildRegion = (
+    a11yStrings: NestedArray<string>,
+    callback: (regionStrings: NestedArray<string>) => void,
+) => {
+    const regionStrings: NestedArray<string> = [];
+    a11yStrings.push(regionStrings);
+    callback(regionStrings);
+};
+
+const handleObject = (
+    tree: AnyParseNode,
+    a11yStrings: NestedArray<string>,
+    atomType: Atom | "normal",
+) => {
+     // Everything else is assumed to be an object...
+    switch (tree.type) {
+        case "accent": {
+            buildRegion(a11yStrings, (a11yStrings) => {
+                buildA11yStrings(tree.base, a11yStrings, atomType);
+                a11yStrings.push("with");
+                buildString(tree.label, "normal", a11yStrings);
+                a11yStrings.push("on top");
+            });
+            break;
+        }
+
+        case "accentUnder": {
+            buildRegion(a11yStrings, (a11yStrings) => {
+                buildA11yStrings(tree.base, a11yStrings, atomType);
+                a11yStrings.push("with");
+                buildString(accentUnderMap[tree.label], "normal", a11yStrings);
+                a11yStrings.push("underneath");
+            });
+            break;
+        }
+
+        case "accent-token": {
+            throw new Error("TODO: accent-token");
```

Comment:
Is it possible to prefix exceptions with something like `KaTeX a11y` to ease tracing back the error?

Reply:
Good idea, will do!